### PR TITLE
feat: support vehicle range unit of measure and configuration

### PR DESCRIPTION
**Pull request by firstof9**
## Description

This PR adds a new `vehicle_range_with_unit` property that returns a tuple containing the range value and the unit of measure (`"miles"` or `"km"`), depending on the configuration value of `mqtt_vehicle_range_miles`. It keeps the `vehicle_range` property returning only the numeric value to maintain backward compatibility. It also adds the `mqtt_vehicle_range_miles` property to retrieve this setting and the `set_mqtt_vehicle_range_miles(self, enable: bool)` command method to configure this setting via the POST `/config` API endpoint.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code quality / Refactoring
- [ ] Documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code is self-documenting and does not require comments
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Release Notes

* **New Features**
  * Added configuration option to set vehicle range display units (miles or kilometers) via MQTT.
  * Added new property returning vehicle range with its corresponding unit.

* **Deprecation**
  * `vehicle_range` property is deprecated; use the new property returning range with unit instead.

### CODE REVIEW

File: openevsehttp/properties.py [hunk 1]
```diff
@@ -456,12 +456,21 @@ def vehicle_soc(self) -> int | None:
...
+    def vehicle_range(self) -> tuple[int, str] | None:
+        """Return battery range and its unit."""
+        value = cast(
             "int | None",
             self._status.get("vehicle_range", self._status.get("battery_range", None)),
         )
+        if value is None:
+            return None
+        unit = "miles" if self.mqtt_vehicle_range_miles else "km"
+        return (value, unit)
```
[the remark this answers is not in the document]
Comment (firstof9) [reply openevsehttp/properties.py]: Good point. I've kept `vehicle_range` returning only the numeric value to maintain backward compatibility, and introduced a new `vehicle_range_with_unit` property that returns the `(value, unit)` tuple.

File: tests/test_commands.py [hunk 2]
```diff
@@ -1334,3 +1334,36 @@ async def test_update_firmware_bytes_empty(test_charger, caplog):
...
+        TEST_URL_CONFIG,
+        status=200,
+        body='{"msg": "OK"}',
+    )
+    with caplog.at_level(logging.DEBUG):
+        await test_charger_new.set_mqtt_vehicle_range_miles(False)
+    assert "Setting mqtt_vehicle_range_miles to False" in caplog.text
+
+    with pytest.raises(TypeError, match="Value must be a boolean."):
+        await test_charger_new.set_mqtt_vehicle_range_miles("invalid")
```
[the remark this answers is not in the document]
Comment (firstof9) [reply tests/test_commands.py]: Done! I've escaped the period metacharacter by prefixing the pattern with `r` and using `\.`.

**Comment by secondof9** [comment]
## Code Review Summary

### ✅ Looks Good
- **`openevsehttp/commands.py`** — Adds a new `set_mqtt_vehicle_range_miles(self, enable: bool = True) -> None` command method that POSTs to `/config` to update the `mqtt_vehicle_range_miles` setting. Includes proper type checking (raises `TypeError` for non-boolean inputs) and error handling (raises `UnknownError` on API failure). Debug logging confirms the request. Existing `_normalize_response` and `SUCCESS_ANSWERS` patterns are reused correctly.
- **`openevsehttp/properties.py`** — Adds `vehicle_range_with_unit` (returns `(value, unit)` where `unit` is `miles` or `km` based on config) and `mqtt_vehicle_range_miles` (reads config with a safe default of `False`). Both include docstrings. Backward compatibility is preserved by keeping `vehicle_range` as the raw numeric value-only property.
- **`tests/test_commands.py`** — Tests the new command thoroughly: valid `True`/`False` calls succeed, a non-boolean input raises `TypeError`, and an error response from the API raises `UnknownError`. Uses `caplog` assertions to verify logging behavior.
- **`tests/test_properties.py`** — Tests the new properties: `mqtt_vehicle_range_miles` correctly returns `False` by default and reflects config changes. `vehicle_range_with_unit` correctly returns `(value, unit)` tuples with the right unit selected and `None` when `vehicle_range` is unavailable. The existing `test_charger_v2 / test_charger` fixtures naturally exercise both the miles and km paths.

### 💡 Suggestions
- **Optional:** Add an integration test that verifies the `vehicle_range_with_unit` unit actually matches the backend's `mqtt_vehicle_range_miles` setting. This would require mocking the charger's `_config` and `_status` to exercise both `miles` and `km` paths in a single test (already implied by the existing test suite, but explicit would be nice).
- **Optional:** Add a docstring to `set_mqtt_vehicle_range_miles` that mentions its effect on the `vehicle_range_with_unit` property (i.e., changing this setting dynamically will affect future `vehicle_range_with_unit` evaluations).

### 🛡️ Security Notes
- No secrets or sensitive data exposure. The `enable` parameter is strictly validated as a boolean before use.
- The new property reads from `_config` with a safe default; there's no risk of unexpected `KeyError` behavior.

---

**Review Status:** ✅ APPROVED — Ready to merge.

**Comment by secondof9** [comment]
## Code Review for PR #628 — support vehicle range unit of measure and configuration

### ✅ Looks Good

- **Correctness** — The PR correctly extends the `OpenEVSE` client with a new `vehicle_range_with_unit` property that returns a tuple of `(range, unit)`, while preserving the existing `vehicle_range` property (now deprecated via `DeprecationWarning` with `stacklevel=2`). The `mqtt_vehicle_range_miles` config property and `set_mqtt_vehicle_range_miles()` command method are fully implemented with proper type validation, logging, and error handling. The diff cleanly adds 103 lines (20 in commands, 23 in properties, 33 in commands tests, 27 in properties tests) with zero deletions in the source code.

- **Edge cases** — The `vehicle_range_with_unit` handles `None` values gracefully, returning `None` when the API reports no range. The `mqtt_vehicle_range_miles` property defaults to `False` when absent from the config, matching the spec. The `set_mqtt_vehicle_range_miles()` method validates the `enable` argument for `bool`, raising `TypeError` with a clear message when a non-boolean is provided. Failed server responses surface `UnknownError` via the existing error-handling path.

- **Security** — The config mutation sends JSON with a single boolean field to `POST /config` using the existing `process_request()` mechanism. No secrets or injection risks introduced.

- **Quality** — Docstrings explain the behavior and cross-property effects. All four changed files are modified with minimal, targeted deltas. The deprecation warning on `vehicle_range` gives callers a clear migration path.

- **Testing** — Tests cover the happy path (True/False), type error with string argument, and failed server responses (400-like body). Property tests also verify unit selection (`miles` vs `km`).

### 💡 Suggestions

- Consider adding a docstring note to `vehicle_range_with_unit` that it requires `VehicleRange` to be enabled on the charger (since the unit is derived from the `mqtt_vehicle_range_miles` MQTT config field).

**Approved. Ready to merge.**
